### PR TITLE
Implement CS checking based on the `WP_CLI_CS` ruleset

### LINKED ISSUE #54: Adopt and enforce new `WP_CLI_CS` standard
We have a new PHPCS standard for WP-CLI called `WP_CLI_CS` (props @jrfnl). It is part of the `wp-cli/wp-cli-tests` package starting with version v2.1.0.

To adopt & enforce this new standard, the following actions need to be taken for this repository:

- [x] Create a PR that adds a custom ruleset `phpcs.xml.dist` to the repository

    - Add `phpcs.xml.dist` file
    - Adapt `.distignore` to ignore `phpcs.xml.dist` & `phpunit.xml.dist`
    - Adapt `.gitignore` to ignore `phpunit.xml`, `phpcs.xml` & `.phpcs.xml`
    - Require version `^2.1` of the `wp-cli/wp-cli-tests` as a dev dependency
- [x] Make any required changes to the code that fail the checks from the above ruleset in separate PRs
- [ ] Merge the ruleset once all required changes have been processed and merged

A sample PR for a simple repository can be seen here: https://github.com/wp-cli/maintenance-mode-command/pull/3/files

Related https://github.com/wp-cli/wp-cli/issues/5201

### PR DESCRIPTION
**Pull request by thrijith**
Add a PHPCS ruleset using the new `WP_CLI_CS` standard.

Fixes #54 

Related https://github.com/wp-cli/wp-cli/issues/5201

### CODE REVIEW

**Review by schlessera — APPROVED** [review]

File: src/Find_Command.php [hunk 1]
```diff
@@ -233,7 +234,7 @@ private function recurse_directory( $path ) {
 		// Don't recurse directories that probably don't have a WordPress installation.
 		if ( ! $this->skip_ignored_paths ) {
 			// Assume base path doesn't need comparison
-			$compared_path = preg_replace( '#^' . preg_quote( $this->base_path ) . '#', '', $path );
+			$compared_path = preg_replace( '#^' . preg_quote( $this->base_path, '/' ) . '#', '', $path );
```
Comment (jrfnl) [inline src/Find_Command.php:237]: Uh.. oh... this is wrong... the delimiter should be passed as the second parameter and in this case, the delimiter is `#`.

Comment (thrijith) [reply src/Find_Command.php:237]: @schlessera, should I send a PR with `#` as the delimiter for `preg_quote`? I don't recall why I added `/`, not sure which is correct here.

Comment (schlessera) [reply src/Find_Command.php:237]: @jrfnl Ah, good catch, thanks for that!

@thrijith Yes, please.

Comment (schlessera) [reply src/Find_Command.php:237]: @thrijith Did you maybe intend to concatenate to add trailing slash here... ?

Comment (schlessera) [reply src/Find_Command.php:237]: Either way, `preg_quote()` should have a `#` as the second argument in this case.

**Pull request merged**